ocp_resources/resource.py

```diff
@@ -970,27 +970,23 @@ def wait_for_status(
                 **PROTOCOL_ERROR_EXCEPTION_DICT,
                 **DEFAULT_CLUSTER_RETRY_EXCEPTIONS,
             },
-            func=self.api.get,
-            field_selector=f"metadata.name=={self.name}",
-            namespace=self.namespace,
+            func=lambda: self.instance.status.phase,
```

Comment:
can we fail here in cases where there's no `status.phase` like "None has no attirbute..."?

Reply:
how about 
```
func=lambda: self.exists,
```
and 
```
if sample:
    current_status = sample.status.phase
```

Reply:
Switch to use `self.exists`